File: lib/modifiers.py
from .constraints import Constraint
from .functions import Two_Value


class Modifier(Constraint):
    ...


class IDENTITY(Two_Value, Modifier):
    def __init__(self, value, first, second) -> None:
        Modifier.__init__(self, value)
        Two_Value.__init__(self, first, second)

    @property
    def name(self):
        return f"{self.value} {super().name}"


class AUTOINCREMENT(Modifier):
    ...

File: prmp_sql_dart/src/modifiers.dart
import './constraints.dart' show Constraint;
import './functions.dart' show Function_Two_Value;

class Modifier extends Constraint {
  Modifier(value) : super(value);
}

class IDENTITY extends Function_Two_Value {
  dynamic value;

  IDENTITY(this.value, first, second) : super(first, second);
  @override
  String get name => '$value ${super.name}';
}

class AUTOINCREMENT extends Modifier {
  AUTOINCREMENT(value) : super(value);
}
